### PR TITLE
bump pydantic version for Contention Classification App

### PR DESCRIPTION
**Pull request by lukey-luke**
## What was the problem?

- updated pydantic version in aim to avoid secrel issues requiring update at unexpected time mid-sprint

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/374

## How does this fix it?[^1]

- I used the recommend `bump-pydantic` package from the [pydantic migration guide](https://docs.pydantic.dev/2.6/migration/)
- Updated syntax for @model_validator manually
- updated requirements.txt

## How to test this PR
- run `pytest`
- test POST request to cc-app via postman, swagger docs, etc

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by tyler-spangler6 — APPROVED** [review]
Approved

File: domain-cc/cc-app/src/python_src/api.py [hunk 1]
```diff
@@ -93,7 +93,9 @@ def log_claim_stats(claim: Claim, classification: Optional[PredictedClassificati
     contention_text = claim.contention_text or ""
     is_in_dropdown = contention_text.strip().lower() in dropdown_values
     is_mapped_text = dropdown_lookup_table.get(contention_text, None) is not None
-    log_contention_text = contention_text if is_mapped_text else "unmapped contention text"
+    log_contention_text = (
+        contention_text if is_mapped_text else "unmapped contention text"
+    )
```
Comment (raudabaugh) [inline domain-cc/cc-app/src/python_src/api.py:98]: Did you setup a new autoformatter?

**Review by raudabaugh — APPROVED** [review]
LGTM

File: domain-cc/cc-app/src/python_src/api.py (hunk 1 above)
Comment (lukey-luke) [reply domain-cc/cc-app/src/python_src/api.py:98]: I didn't change formatting. Precommit hooks are defined [here](https://github.com/department-of-veterans-affairs/abd-vro/blob/a4003bf20f4e2558c704f96009203eb8b6386b8e/domain-cc/.pre-commit-config.yaml#L27). Perhaps I had committed w/ `--no-verify` prior, not sure

**Pull request merged**